### PR TITLE
[user-authn] Improve groups migration

### PR DESCRIPTION
**Pull request by nabokihms**
## Description
* Run migration only once
* Slugify names
* Add the group name field to the output table

## Why do we need it, and what problem does it solve?
The previous version of the migration can fail or overwrite manual changes.

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: user-authn
type: fix
summary: Improve the groups migration (run only once and slugify group names)
```

### CODE REVIEW

File: modules/150-user-authn/hooks/migration_groups_from_user_to_group_object.go [hunk 1]
```diff
@@ -75,33 +90,72 @@ func migrationGroups(input *go_hook.HookInput, dc dependency.Container) error {
...
 		}
-		newDexGroup := &DexGroup{
+		dexGroup := &DexGroup{
 			TypeMeta: metav1.TypeMeta{
 				Kind:       DexGroupKind,
 				APIVersion: DexGroupAPIVersion,
 			},
 			ObjectMeta: metav1.ObjectMeta{
-				Name: strings.ToLower(groupName),
+				Name: metadataNameify(groupName),
```
Comment (shvgn) [inline modules/150-user-authn/hooks/migration_groups_from_user_to_group_object.go:99]: 👍😅

File: modules/150-user-authn/hooks/migration_groups_from_user_to_group_object_test.go [hunk 2]
```diff
@@ -61,55 +88,94 @@ metadata:
 spec:
   email: user@example.com
   groups:
+  - admins
```
Comment (shvgn) [inline modules/150-user-authn/hooks/migration_groups_from_user_to_group_object_test.go:91]: I believe we should preserve group names which were valid metadata.names from the beginning. Thus we would not punish users who were lucky to name groups according to new concerns. They would not need to rename groups if they were named well.

E.g. 

- admins → admins
- Admins → admins-(somehash)
- -Admins → admins-(anotherhash)

Comment (nabokihms) [reply modules/150-user-authn/hooks/migration_groups_from_user_to_group_object_test.go:91]: Good point, fixed it.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/static'...

**Review by shvgn — APPROVED** [review]

**Pull request merged**